File: 2492.ts
function minScore(n: number, roads: number[][]): number {
  const graph = Array.from({ length: n + 1 }, (): number[][] => []);
  roads.forEach(([u, v, d]) => {
    graph[u].push([v, d]);
    graph[v].push([u, d]);
  });

  const visited = new Array<boolean>(n + 1).fill(false);
  let ans = Infinity;
  const dfs = (v: number): void => {
    visited[v] = true;
    for (const [u, d] of graph[v]) {
      ans = Math.min(ans, d);
      if (!visited[u]) {
        dfs(u);
      }
    }
  };
  dfs(1);
  return ans;
}

File: 2492.py
class Solution(object):
    def minScore(self, n, roads):
        graph = [[] for _ in range(n + 1)]
        for a, b, distance in roads:
            graph[a].append((b, distance))
            graph[b].append((a, distance))

        min_score = float('inf')
        visited = [False] * (n + 1)
        queue = deque()
        queue.append(1)
        visited[1] = True

        while queue:
            city = queue.popleft()
            for neighbor, distance in graph[city]:
                if distance < min_score:
                    min_score = distance
                if not visited[neighbor]:
                    visited[neighbor] = True
                    queue.append(neighbor)

        return min_score

